### PR TITLE
Add permissions to allow Jenkins to run an ssm document.

### PR DESCRIPTION
**Pull request by MancunianSam**
This is being used to delete the bastion db user when the bastion is
destroyed.

### CODE REVIEW

File: modules/environment-roles/templates/terraform_scripts_policy.json.tpl [hunk 1]
```diff
@@ -62,23 +62,40 @@
...
+        "ssm:UpdateDocumentDefaultVersion"
       ],
       "Resource": [
         "arn:aws:dynamodb:eu-west-2:${account_id}:table/tdr-terraform-state-lock-scripts",
         "arn:aws:iam::${account_id}:instance-profile/bastion",
         "arn:aws:iam::${account_id}:role/BastionEC2Role${title_environment}",
+        "arn:aws:iam::${account_id}:role/TDRBastionAccessDbRole${title_environment}",
+        "arn:aws:iam::${account_id}:policy/TDRBastionAccessDbPolicy${title_environment}",
+        "arn:aws:iam::${account_id}:policy/TDRBastionAssumeDbRolePolicy${title_environment}",
+        "arn:aws:iam::${account_id}:role/TDRJenkinsRunDocumentRoleIntg",
```
Comment (TomJKing) [inline modules/environment-roles/templates/terraform_scripts_policy.json.tpl:85]: Should "Intg" be hardcoded to the the `TDRJenkinsRunDocumentRoleIntg` and `TDRJenkinsRunDocumentPolicyIntg` resource names?

File: modules/specific-environment-permissions/environment_permissions.tf [hunk 2]
```diff
@@ -85,6 +85,22 @@ resource "aws_iam_role_policy_attachment" "jenkins_role_lambda_attachment" {
   role       = aws_iam_role.jenkins_lambda_assume_role.name
 }
 
+
+resource "aws_iam_role" "jenkins_run_ssm_role" {
+  name               = "TDRJenkinsRunSsmRole${local.env_title_case}"
+  assume_role_policy = templatefile("${path.module}/templates/ecs_assume_role_policy.json.tpl", {})
+}
+
+resource "aws_iam_policy" "tdr_jenkins_run_ssm_policy" {
```
Comment (TomJKing) [inline modules/specific-environment-permissions/environment_permissions.tf:94]: Is the "tdr_" prefix needed for the resource name? Not used elsewhere.

File: modules/specific-environment-permissions/templates/jenkins_assume_role.json.tpl [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": "sts:AssumeRole",
+      "Resource": [
+        "arn:aws:iam::${account_id}:role/TDRJenkinsRunDocumentRole${environment}"
+      ]
+    }
+  ]
+}
```
Comment (TomJKing) [inline modules/specific-environment-permissions/templates/jenkins_assume_role.json.tpl:12]: Missing line break

**Review by TomJKing — COMMENTED** [review]
Looks good. Couple of minor comments. Assume these permissions are needed before deploying PRs:

- https://github.com/nationalarchives/tdr-scripts/pull/5; and
- https://github.com/nationalarchives/tdr-terraform-modules/pull/88

File: modules/environment-roles/templates/terraform_scripts_policy.json.tpl (hunk 1 above)
Comment (MancunianSam) [reply modules/environment-roles/templates/terraform_scripts_policy.json.tpl]: Ah good spot, I thought I'd got all the hard coded ones. I've changed it.

File: modules/specific-environment-permissions/environment_permissions.tf (hunk 2 above)
Comment (MancunianSam) [reply modules/specific-environment-permissions/environment_permissions.tf:94]: True, I've removed it.

File: modules/specific-environment-permissions/templates/jenkins_assume_role.json.tpl (hunk 3 above)
Comment (MancunianSam) [reply modules/specific-environment-permissions/templates/jenkins_assume_role.json.tpl]: I've added one in.

**Review by TomJKing — APPROVED** [review]

**Pull request merged**